import_from_csv to auto_detect_line_endings

bdtech opened the pull request:
**Bug**: Line endings caused issues for importing CSV that resulted in incomplete imports. 

**Fix**: This allows PHP to interoperate with Macintosh systems (including \r line endings from older Macs).

Note: The PHP [documentation](https://www.php.net/manual/en/filesystem.configuration.php#ini.auto-detect-line-endings) mentions the possibility of a minor performance hit. Since the CSV is only two columns (not too data-intensive bytes wise), while running on modern hardware CPU's, I'm not anticipating any bottlenecks at all.

**Tested**: Import tested successfully with various line endings and imported the entire file accurately. I also tested with a CSV provided by Gary since that was a suitable test case.

Related Issue #52 @GaryJones

bdtech committed:
import_from_csv to auto_detect_line_endings

Line endings caused issues for importing CSV that resulted in incomplete imports. This allows PHP to interoperate with Macintosh systems (including \r line endings from older Macs).

Note: The php documentation mentions the possibility of a minor performance hit. Since the CSV is only two columns (not too data intenstive bytes wise), while running on modern hardware CPU's, i'm not anticipating any bottlenecks at all.

bdtech committed:
Merge remote-tracking branch 'upstream/develop' into fix-importcsv-line-endings

rebeccahum commented on includes/class-wpcom-legacy-redirector-cli.php:
```diff
@@ -257,6 +257,8 @@ function import_from_csv( $args, $assoc_args ) {
 			WP_CLI::line( 'Processing...' );
 		}
 
+		ini_set('auto_detect_line_endings',TRUE);
```
This is minor, it would be nice to space code per the WordPress PHP Coding Standards: https://make.wordpress.org/core/handbook/best-practices/coding-standards/php/#space-usage.

Also, should we check that it isn't `TRUE` already before setting it and if so, should we set it back to its original value after processing?

bdtech replied:
@rebeccahum Good catch, I was coding with Sublime Text instead of PHP Storm to save some resources. I've applied the code formatting. 👍 

Since this is an internal CLI function limited in scope to importing the CSV I don't believe its necessary. Also, the PHP documentation does not mention that it will be a "breaking" change. Don't see much risk here. What do you think?

rebeccahum replied:
@bdtech Correct, it is an internal CLI function but it is also used on local test environments and leaving it on could interfere with other script executions.  Additionally, do you think there could be any performance implications with leaving it on?

GaryJones replied:
What makes you think this is an _internal_ CLI?

I've got this plugin running on a local environment, outside of VIP, and I can use the CLI commands just fine. We must remember that this plugin could be active on SiteGround, WPEngine, Pantheon, or other self-hosted hosts which allow SSH + WP-CLI access.

bdtech replied:
@GaryJones Good point let me clarify. I meant isolated to the plugin itself. This only fires when the CSV is being loaded through our CLI function.

GaryJones replied:
I think you're right. I can't think of a case where the current script execution would be doing something else in parallel.

That said, I'd like to see a comment that explained that the change is only for the current request, and automatically restores to the default at the end of the execution.

bdtech replied:
@rebeccahum I've added context regarding the scope of this change.

rebeccahum reviewed (changes requested):
Thanks for the contribution @bdtech, please see my input:

bdtech committed:
fix spacing

reformat and apply code style for spacing/case

bdtech committed:
auto_detect_line_endings: add note

Add comment explaining the ini override only occurs temporarily for this CLI function call. This will not persist or impact any other requests (or page loads).
